raytracing/matrix.py

```diff
@@ -378,6 +380,62 @@ def _waitForFile(self, filePath, timeout=5):
             oldSize = currentSize
         time.sleep(0.1)
 
+    def traceManyThroughInMultiprocess(self, inputRays, processes=4):
+        """
+        Naive implementation of multiprocessing on `inputRays`.
+        """
+        with multiprocessing.Pool(processes=processes) as pool:
+            outputRays = pool.map(self.traceThrough, inputRays)
+        return Rays(rays=outputRays)
+
+    def traceManyThroughInMultiprocessChunks(self, inputRays, processes=4):
```

Comment:
I find this code to difficult to read for non-experts even though I like the idea of using `traceManyTrough()` instead in `traceTrough()`. Also, if we create new dynamic classes, this will fail because e we check for a specific class.  This is not proper object-oriented programming because implementation details appear in the function.  It is better to avoid any specifics.

I found simple code to split my own array into smaller arrays easily at https://stackoverflow.com/questions/752308/split-list-into-smaller-lists-split-in-half

```
        theExplicitList = list(inputRays) 
        manyInputRays = [theList[i::processes] for i in range(processes)]
```

so I do that in the main function now, which I renamed again "inParallel".